py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,797 @@
+   - {snow}: Contains the formatted result of format_snow.
+   - {sunrise}: Contains the formatted result of format_sunrise.
+   - {sunset}: Contains the formatted result of format_sunset.
+   - {temp}: Contains the formatted result of format_temp.
+   - {wind}: Contains the formatted result of format_wind.
+
+
+Color options:
+    color_neg_20: Color for a Fahrenheit temperature of -20°.
+        Defaults to a magenta.
+        (default '#FF00FF')
+    color_neg_60: Color for a Fahrenheit temperature of -60°.
+        Defaults to a deep purple.
+        (default '#6B006B')
+    color_pos_120: Color for a Fahrenheit temperature of 120°.
+        Defaults to a white.
+        (default '#FFFFFF')
+    color_pos_30: Color for a Fahrenheit temperature of 30°.
+        Defaults to a cyan.
+        (default '#00FFFF')
+    color_pos_40: Color for a Fahrenheit temperature of 40°.
+        Defaults to a green.
+        (default '#7FFF00')
+    color_pos_50: Color for a Fahrenheit temperature of 50°.
+        Defaults to a yellow.
+        (default '#7FFF00')
+    color_pos_70: Color for a Fahrenheit temperature of 70°.
+        Defaults to a orange.
+        (default '#FF9900')
+    color_pos_90: Color for a Fahrenheit temperature of 90°.
+        Defaults to a red.
+        (default '#FF0000')
+    color_zero: Color for a Fahrenheit temperature of zero.
+        Defaults to a blue.
+        (default '#0000FF')
+
+
+Example configuration:
+```
+weather_owm {
+  api_key = '...'
+
+  icons {
+    i200 = "☔"
+    i230_i232 = "🌧"
+
+    clouds = "☁"
+  }
+
+  format = '{icon}: {temp}, {forecast}'
+  forecast_days = 3
+}
+```
+Outputs: 🌫: ○: 59°, ⛅ ☼ 🌧`
+- Currently foggy, 59° F outside, with forecast of cloudy tomorrow, sunny the
+  next day, then rainy
+
+
+@author alexoneill
+@licence MIT
+'''
+
+import datetime
+
+
+# API information
+OWM_API = '2.5'
+OWM_CURR_ENDPOINT = 'http://api.openweathermap.org/data/%s/weather?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s'
+OWM_FUTURE_ENDPOINT = 'http://api.openweathermap.org/data/%s/forecast?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s&cnt=%%d'
+IP_ENDPOINT = 'http://ip-api.com/json'
+
+# Paths of information to extract from JSON
+IP_LAT = '//lat'
+IP_LNG = '//lon'
+OWM_WEATHER_ICON = '//weather:0/id'
+OWM_CLOUD_COVER = '//clouds/all'
+OWM_RAIN = '//rain/3h'
+OWM_SNOW = '//snow/3h'
+OWM_WIND = '//wind'
+OWM_HUMIDITY = '//main'
+OWM_PRESSURE = '//main'
+OWM_TEMP = '//main'
+OWM_SUNRISE = '//sys/sunrise'
+OWM_SUNSET = '//sys/sunset'
+OWM_DESC = '//weather:0/main'
+OWM_DESC_LONG = '//weather:0/description'
+
+
+class OWMException(Exception):
+    pass
+
+
+class Py3status:
+
+    api_key = None
+    cache_timeout = 600
+    color_neg_20 = '#FF00FF'
+    color_neg_60 = '#6B006B'
+    color_pos_120 = '#FFFFFF'
+    color_pos_30 = '#00FFFF'
+    color_pos_40 = '#7FFF00'
+    color_pos_50 = '#7FFF00'
+    color_pos_70 = '#FF9900'
+    color_pos_90 = '#FF0000'
+    color_zero = '#0000FF'
```

Comment:
These can probably be better handled using the thresholds functionality.  Also bear in mind that much of the world uses a different temperature scale.